sqlglot/dialects/tsql.py

```diff
@@ -643,6 +659,7 @@ class Generator(generator.Generator):
             exp.Min: min_or_least,
             exp.NumberToStr: _format_sql,
             exp.Select: transforms.preprocess([transforms.eliminate_distinct_on]),
+            exp.SetItem: lambda self, e: f"{self.sql(e.this.left)}{' ' if isinstance(e.this.left, exp.Identifier) else ' = '}{self.sql(e.this.right)}",
```

Comment:
@dmoore247 can you explain the logic behind this condition? why do you check for an identifier?

Reply:
`SET @var =` and `SET KEY VALUE` need to be treated differently. I happened to noticed one was parsed as an IDENTIFIER and the other was not. Any advice?

Reply:
Found example where this logic isn't working... `SET @user OFF`
https://github.com/tobymao/sqlglot/blame/edd3804410d43904da12fcc65ffe3bb16774cf0f/tests/fixtures/identity.sql#L187

Back to drawing board.